sqldelight-gradle-plugin/src/test/fixtures/case-expression-type/expected/com/test/TestModel.java

```diff
@@ -52,29 +52,29 @@ public T map(@NonNull Cursor cursor) {
     }
   }
 
-  class Marshal<T extends Marshal<T>> {
+  final class Marshal {
     protected ContentValues contentValues = new ContentValues();
 
-    public Marshal() {
+    private Marshal() {
     }
 
-    public Marshal(TestModel copy) {
+    private Marshal(TestModel copy) {
```

Comment:
All I see is this
![](http://www.crystalworlds.com/newsletters/images/hallway.jpg)
except the walls say `Marhsal.access$000` everywhere